addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactHandler.java

```diff
@@ -473,51 +406,67 @@ public void run() {
         @Override
         public void run() {
             try {
+                long stamp = System.currentTimeMillis();
+                if (!InetAddress.getByName(((String) getConfig().get(IP_ADDRESS))).isReachable(PING_TIME_OUT)) {
+                    logger.debug("Ping timed out after '{}' milliseconds", System.currentTimeMillis() - stamp);
+                    logger.trace("Disconnecting the datagram channel '{}'", datagramChannel);
+                    try {
+                        datagramChannel.close();
+                    } catch (IOException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "An exception occurred while closing the channel");
+                    }
 
-                String command = "report 1";
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "A ping timeout occurred");
+                    onConnectionLost();
+                } else {
 
-                ByteBuffer byteBuffer = ByteBuffer.allocate(command.getBytes().length);
-                try {
-                    byteBuffer.put(command.getBytes("ASCII"));
-                    onWritable(byteBuffer, datagramChannel);
-                } catch (UnsupportedEncodingException | NumberFormatException e) {
-                    logger.error("An exception occurred while polling the KEBA KeContact P20 for '{}': {}",
-                            getThing().getUID(), e.getMessage());
-                }
+                    String command = "report 1";
 
-                Thread.sleep(REPORT_INTERVAL);
+                    ByteBuffer byteBuffer = ByteBuffer.allocate(command.getBytes().length);
+                    try {
+                        byteBuffer.put(command.getBytes("ASCII"));
+                        onWritable(byteBuffer, datagramChannel);
+                    } catch (UnsupportedEncodingException | NumberFormatException e) {
```

Comment:
If it makes sense that the code blocks have their own try-catch it might be a good idea to extract 1 method, because there also seems to some duplication in the fetching of the 3 reports.